CMakeLists.txt

```
@@ -139,6 +139,16 @@ set(_SHARED_LIB_SUFFIX
 set(_SUNDIALS_ALIAS_TARGETS ""
     CACHE INTERNAL "" FORCE)
 
+# We default to release builds
+set(_DEFAULT_CMAKE_BUILD_TYPE Release)
```

I think xSDK still requires debug to be the default build type
```suggestion
if(USE_XSDK_DEFAULTS)
  set(_DEFAULT_CMAKE_BUILD_TYPE Default)
else()
  set(_DEFAULT_CMAKE_BUILD_TYPE Release)
endif()
```
and the following code can be remove from `SundialsBuildOptionsPre.cmake`
```
if(USE_XSDK_DEFAULTS)
  message(STATUS "Enabling xSDK defaults:")

  # set the CMake build type, SUNDIALS does not set a build type by default
  if(NOT CMAKE_BUILD_TYPE)
    message(STATUS "  Setting build type to Debug")
    set(DOCSTR "Choose the type of build: None Debug Release RelWithDebInfo MinSizeRel")
    set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "${DOCSTR}" FORCE)
  endif()
endif()
```

---

xSDK does not require a default anymore (and has not for awhile). It just requires that a `Debug` option exists (see [policy M16](https://github.com/xsdk-project/xsdk-community-policies/blob/master/package_policies/M16.md)).

---

I like `RelWithDebInfo` as a default.

---

> xSDK does not require a default anymore (and has not for awhile). It just requires that a `Debug` option exists (see [policy M16](https://github.com/xsdk-project/xsdk-community-policies/blob/master/package_policies/M16.md)).

Ah, good. In that case we can just remove the block of code noted above in `SundialsBuildOptionsPre.cmake`. 

> If we want to match what Spack does by default, we could make our default mode `RelWithDebInfo` instead of `Release`.

`RelWithDebInfo` sounds good